### PR TITLE
PMM-7591 Fix alertmanager configuration updates

### PR DESCRIPTION
**Pull request by artemgavrilov**
[PMM-7591](https://jira.percona.com/browse/PMM-7591)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1554

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Comment by artemgavrilov** [comment]
FB is on the way

File: services/server/server_test.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func TestServer(t *testing.T) {
 		mvmalert.Test(t)
 		mvmalert.On("RequestConfigurationUpdate").Return(nil)
 
+		malertmanager := new(mockAlertmanagerService)
+		malertmanager.Test(t)
+		malertmanager.On("RequestConfigurationUpdate").Return(nil)
```
Comment (oter) [inline services/server/server_test.go:59]: maybe check that `RequestConfigurationUpdate` was called?

**Review by oter — APPROVED** [review]

**Review by oter — COMMENTED** [review]
please change base branch from `PMM-2.0` to `release/2.16`

**Comment by artemgavrilov** [comment]
> please change base branch from `PMM-2.0` to `release/2.16`

This bugfix should go to the 2.15 release

File: services/server/server_test.go (hunk 1 above)
Comment (BupycHuk) [reply services/server/server_test.go:59]: As far as I remember this already should do that.

**Review by BupycHuk — APPROVED** [review]

File: services/server/server_test.go (hunk 1 above)
Comment (oter) [reply services/server/server_test.go:59]: actually no, you should specify number of calls to check and then call `mock.AssertExpectationsForObjects()`

example:

```
func TestMock(t *testing.T) {
	mvmdb := new(mockPrometheusService)
	mvmdb.Test(t)
	mvmdb.On("RequestConfigurationUpdate").Return(nil).Once()

	mock.AssertExpectationsForObjects(t, mvmdb)
}
```

Comment (BupycHuk) [reply services/server/server_test.go:59]: Yeah, you are right, we don't call AssertExpectations here

Comment (artemgavrilov) [reply services/server/server_test.go:59]: I don't see an easy and simple way to add such checks now.

Comment (BupycHuk) [reply services/server/server_test.go:59]: In teardown? We have several places where we do it or do you mean that this method is not called in all places?

Comment (artemgavrilov) [reply services/server/server_test.go:59]: Yes, it's called not in all subtests

**Pull request merged**